jsx-factory.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import('./lib/core.js').Element} Element
+ * @typedef {import('./lib/core.js').Root} Root
+ * @typedef {import('./lib/core.js').HResult} HResult
+ * @typedef {import('./lib/core.js').HArrayChild} HArrayChild
+ * @typedef {import('./lib/core.js').HProperties} HProperties
+ * @typedef {import('./lib/core.js').HPropertyValue} HPropertyValue
+ * @typedef {import('./lib/core.js').HStyle} HStyle
+ * @typedef {import('./lib/core.js').core} Core
+ *
+ * @typedef {Record<string, unknown>} JSXProps
+ * @typedef {(props: JSXProps) => HResult } HFn
```

Comment:
```suggestion
 */

/**
 * @typedef {Record<string, unknown>} JSXProps
 * 
 * @callback HFn
 *   Component.
 * @param {JSXProps} props
 *   Props (includes `children`).
 * @returns {HResult}
 *   Result.
```

- Separate block for local definitions
- Docs params/result